sdk/keyvault/Azure.Security.KeyVault.Keys/src/RsaKeyCreateOptions.cs

```diff
@@ -12,22 +12,23 @@ namespace Azure.Security.KeyVault.Keys
 {
     public class RsaKeyCreateOptions : KeyCreateOptions
     {
+        public string Name { get; set; }
+        public KeyType KeyType { get; private set; }
         public int KeySize { get; set; }
-        public KeyType KeyType { get; set; }
+        public bool Hsm { get; set; }
```

Comment:
>public bool Hsm { get; set; } [](start = 8, length = 29)

Hsm should also not be settable as changing it would impact the KeyType

Reply:
changed